Fix email notification link url

lchanmann opened the pull request:
## Ref
- https://nanopay.atlassian.net/browse/NP-3111

## Changes
- Use group.url for notification email link
- Load theme via user spid when the loaded theme's (via domain) spid is not same as user's
- Reload theme when the context theme is available for use in EmailsUtility but the theme's spid is not same as user's

lchanmann committed:
Fix email notification link url

jlhughes commented on src/foam/nanos/notification/NotificationSetting.js:
```diff
@@ -67,21 +67,8 @@ foam.CLASS({
         { name: 'notification', type: 'foam.nanos.notification.Notification' }
       ],
       javaCode: `
-        X userX = x;
```
@lchanmann why are you removing this? 
I intentionally added it recently, so all downstream - from notificationSettings - logic didn't have to worry about this.

lchanmann replied:
Reverted in https://github.com/foam-framework/foam2/pull/4617/commits/816722dda2f442fd04c5ed9425fe5917936c42b5.

jlhughes reviewed (commented):
.

lchanmann committed:
Add condition to load theme via spid when the loaded theme (via domain) is for a different spid than that of the user

lchanmann committed:
Add condition to reload theme when the context theme is for a different spid than that of the user

lchanmann committed:
Revert NotificationSetting.doNotify

lchanmann committed:
Merge branch 'master' into NP-3111/FixEmailNotificationLinkUrl

# Conflicts:
#	src/foam/nanos/theme/Themes.js

lchanmann committed:
Fix client-side loading group theme

jlhughes commented on src/foam/nanos/theme/Themes.js:
```diff
@@ -196,17 +203,6 @@ Later themes:
         }
       }
 
-      if ( theme == null ) {
```
Why did you remove this?

lchanmann replied:
It was moved up to line 178.

jlhughes commented on src/foam/nanos/theme/Themes.js:
```diff
@@ -174,9 +175,15 @@ Later themes:
         }
       }
 
+      if ( theme == null ) {
```
Why did you add this back?

lchanmann replied:
It allows loading group theme when there is no spid theme.

lchanmann committed:
Revert theme fallback logic

lchanmann committed:
Merge branch 'master' into NP-3111/FixEmailNotificationLinkUrl

lchanmann committed:
Throw when spid is empty before put instead of when getting spid

as it prevents the system user from putting other spid data to DAO when no explicit defaultSpid

lchanmann committed:
Revert "Throw when spid is empty before put instead of when getting spid"

This reverts commit 87986afc

lchanmann committed:
Merge branch 'master' into NP-3111/FixEmailNotificationLinkUrl
